examples/reshare/src/validator.rs

```diff
@@ -74,7 +78,10 @@ where
     );
     p2p_cfg.mailbox_size = MAILBOX_SIZE;
 
-    let (mut network, oracle) = discovery::Network::new(context.with_label("network"), p2p_cfg);
+    let (mut network, mut oracle) = discovery::Network::new(context.with_label("network"), p2p_cfg);
+
+    // Register all possible peers
+    oracle.update(0, peer_config.participants.clone()).await;
```

Comment:
We want to register these by epoch?

Reply:
I think the way this is working now is that you have a "known universe" of possible peers across all epochs?